test/cmocka/src/lib/fast-get/fast-get-tests.c

```diff
@@ -0,0 +1,176 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2024 Intel Corporation. All rights reserved.
+//
+// Author: Jyri Sarha <jyri.sarha@linux.intel.com>
+
+#include <sof/lib/fast-get.h>
+#include <rtos/sof.h>
+#include <rtos/alloc.h>
+#include <sof/lib/mm_heap.h>
+#include <sof/lib/memory.h>
+#include <sof/common.h>
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <cmocka.h>
+#include <assert.h>
+
+static int testdata[33][100] = {
+	{
+	1, 2, 3, 4, 5, 6, 7, 9, 0,
+	1, 2, 3, 4, 5, 6, 7, 9, 0,
+	1, 2, 3, 4, 5, 6, 7, 9, 0,
+	1, 2, 3, 4, 5, 6, 7, 9, 0,
+	1, 2, 3, 4, 5, 6, 7, 9, 0,
+	1, 2, 3, 4, 5, 6, 7, 9, 0,
+	1, 2, 3, 4, 5, 6, 7, 9, 0,
+	1, 2, 3, 4, 5, 6, 7, 9, 0,
+	1, 2, 3, 4, 5, 6, 7, 9, 0,
+	1, 2, 3, 4, 5, 6, 7, 9, 0,
+	},
+	{ 2 },
+	{ 3 },
+	{ 4 },
+	{ 5 },
+	{ 6 },
+	{ 7 },
+	{ 8 },
+	{ 9 },
+	{ 10 },
+	{ 11 },
+	{ 12 },
+	{ 13 },
+	{ 14 },
+	{ 15 },
+	{ 16 },
+	{ 17 },
+	{ 18 },
+	{ 19 },
+	{ 20 },
+	{ 21 },
+	{ 23 },
+	{ 24 },
+	{ 25 },
+	{ 26 },
+	{ 27 },
+	{ 28 },
+	{ 29 },
+	{ 30 },
+	{ 31 },
+	{ 32 },
+	{ 33 },
+};
+
+static void test_simple_fast_get_put(void **state)
+{
+	void *ret;
+
+	(void)state; /* unused */
+
+	ret = fast_get(testdata[0], sizeof(testdata[0]));
+
+	assert(!memcmp(ret, testdata[0], sizeof(testdata[0])));
+
+	fast_put(ret);
+}
+
+static void test_over_32_fast_gets_and_puts(void **state)
+{
+	void *copy[ARRAY_SIZE(testdata)];
+	int i;
+	(void)state; /* unused */
+
+	for (i = 0; i < ARRAY_SIZE(copy); i++)
+		copy[i] = fast_get(testdata[i], sizeof(testdata[0]));
+
+	for (i = 0; i < ARRAY_SIZE(copy); i++)
+		assert(!memcmp(copy[i], testdata[i], sizeof(testdata[0])));
+
+	for (i = 0; i < ARRAY_SIZE(copy); i++)
+		fast_put(copy[i]);
+}
+
+static void test_fast_get_refcounting(void **state)
+{
+	void *copy[2][ARRAY_SIZE(testdata)];
+	int i;
+	(void)state; /* unused */
+
+	for (i = 0; i < ARRAY_SIZE(copy[0]); i++)
+		copy[0][i] = fast_get(testdata[i], sizeof(testdata[0]));
+
+	for (i = 0; i < ARRAY_SIZE(copy[0]); i++)
+		copy[1][i] = fast_get(testdata[i], sizeof(testdata[0]));
```

Comment:
you could also test unmatched allocation - with a wrong size. Also you could add a test for too large an allocation - don't think `malloc()` will succeed a 1GiB allocation even if it's available, IIRC there was a limit. But maybe I'm wrong. This can be an incremental PR, this test set is already quite good!